### PR TITLE
Update 'user access' endpoints to be more flexible 

### PR DESCRIPTION
**Pull request by tomsmyers**
this change has two parts

1. change the permissions on the existing `/offenders/crn/{crn}/userAccess` endpoint to allow it to be called with delius user tokens (ROLE_PROBATION).
2. add a new endpoint which allows specifying the username in the url to allow regular client credentials granted tokens the ability to use the endpoint without requesting a new 'username' scoped token.

### CODE REVIEW

**Review by andymarke — DISMISSED** [review]
LGTM

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 1]
```diff
@@ -614,14 +614,27 @@ public SentenceStatus getSentenceStatusBySentenceId(
             @ApiResponse(code = 403, message = "User is restricted from access to offender", response = AccessLimitation.class),
             @ApiResponse(code = 404, message = "No such offender, or no such User (see body for detail)")
     })
+    @PreAuthorize("hasRole('ROLE_COMMUNITY') || hasRole('ROLE_PROBATION')")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:617]: can do `hasAnyRole` such as https://github.com/ministryofjustice/hmpps-auth/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserRolesController.kt#L84

Comment (tomsmyers) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java]: nice - thanks. updated.

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**